source/include/FreeRTOSIPConfigDefaults.h

```diff
@@ -136,19 +136,51 @@
 
 /*---------------------------------------------------------------------------*/
 
+/*===========================================================================*/
+/*                                  MACROS                                   */
+/*===========================================================================*/
+/*---------------------------------------------------------------------------*/
+/*===========================================================================*/
+/*---------------------------------------------------------------------------*/
+/*===========================================================================*/
+/*                           FreeRTOSConfig CHECKS                           */
+/*===========================================================================*/
+
+/*---------------------------------------------------------------------------*/
+
 /*
  * pdFREERTOS_ERRNO_EAFNOSUPPORT
  *
  * Address family not supported by protocol.
  *
- * Note: Now included in FreeRTOS-Kernel/projdefs.h, so this serves as a
- * temporary kernel version check. To be removed in a future version.
+ * Note: pdFREERTOS_ERRNO_EAFNOSUPPORT is now included in
+ * FreeRTOS-Kernel/projdefs.h, defined here for backwards compatibility.
  */
 
+#ifndef pdFREERTOS_ERRNO_EAFNOSUPPORT
+    #define pdFREERTOS_ERRNO_EAFNOSUPPORT    97
+#endif
+
+#if ( INCLUDE_vTaskDelay == 0 )
+    #error INCLUDE_vTaskDelay must be set to 1
+#endif
+
+#if ( INCLUDE_xTaskGetCurrentTaskHandle == 0 )
+    #error INCLUDE_xTaskGetCurrentTaskHandle must be set to 1
+#endif
+
+#if ( configSUPPORT_DYNAMIC_ALLOCATION == 0 )
+    #error configSUPPORT_DYNAMIC_ALLOCATION must be set to 1
+#endif
```

Comment:
Yes, this is good. We can remove this once https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/issues/449 has been resolved.